plugins/polio/js/src/domains/LQAS-IM/shared/RefreshLqasData.tsx

```diff
@@ -0,0 +1,151 @@
+import React, {
+    FunctionComponent,
+    useCallback,
+    useEffect,
+    useState,
+} from 'react';
+import { useSafeIntl, LoadingSpinner } from 'bluesquare-components';
+import moment from 'moment';
+import { Box, Button } from '@material-ui/core';
+import RefreshIcon from '@material-ui/icons/Refresh';
+import { useQueryClient } from 'react-query';
+import {
+    Task,
+    TaskApiResponse,
+} from '../../../../../../../hat/assets/js/apps/Iaso/domains/tasks/types';
+import { useGetLatestLQASIMUpdate } from '../../../hooks/useGetLatestLQASIMUpdate';
+import {
+    useCreateTask,
+    useTaskMonitor,
+} from '../../../../../../../hat/assets/js/apps/Iaso/hooks/taskMonitor';
+import MESSAGES from '../../../constants/messages';
+
+type Props = {
+    countryId?: number;
+    category: string;
+};
+
+const LQAS_TASK_ENDPOINT = '/api/polio/tasks/refreshlqas/';
+
+const useLastUpdate = (
+    lastUpdate: Task<any>,
+): { message: string; updateStatus: string } => {
+    const { formatMessage } = useSafeIntl();
+    const result = { message: '', updateStatus: '' };
+    if (lastUpdate?.status === 'RUNNING') {
+        result.message = formatMessage(MESSAGES.ongoing);
+    }
+    if (lastUpdate?.status === 'SUCCESS' && Boolean(lastUpdate?.ended_at)) {
+        result.message = `${moment
+            .unix(lastUpdate.ended_at as number)
+            .format('LTS')}`;
+    }
+    if (lastUpdate?.status) {
+        const updateStatus =
+            MESSAGES[lastUpdate.status] &&
+            formatMessage(MESSAGES[lastUpdate.status])
+                ? formatMessage(MESSAGES[lastUpdate.status])
+                : '';
+        result.updateStatus = updateStatus;
+    }
+    return result;
+};
+
+export const RefreshLqasData: FunctionComponent<Props> = ({
+    countryId,
+    category,
+}) => {
+    const taskUrl = category === 'lqas' ? LQAS_TASK_ENDPOINT : undefined;
+    const { formatMessage } = useSafeIntl();
+    const [taskId, setTaskId] = useState<number>();
+    const [lastTaskStatus, setlastTaskStatus] = useState<string | undefined>();
+    const queryClient = useQueryClient();
+    const { data: isDataUpdating, isFetching: isFetchingTaskStatus } =
+        useTaskMonitor({
+            taskId,
+            endpoint: taskUrl,
+            invalidateQueries: [category, 'get-latest-task-run'],
+        });
```

Comment:
It needs to be tested, but it sounds about right. Also, since the loading state would be inferred from the latest running task, if we land on the page and another task in running, the button would be disabled, so that would be an, improvement of the feature.
Nice catch!

Reply:
Nice! :D I noticed because that's actually already sort of the situation when you click the button and then refresh the page. You still get the correct state because of the continuous fetching of the last task.

Reply:
If it's OK for you, I'll spin it in another ticket, so I can change it without rush